src/universal/modules/welcome/components/WelcomeWizardForms/Step3InviteTeam.js

```diff
@@ -51,61 +44,102 @@ export default class Step3InviteTeam extends Component {
         label: email.name ? `"${email.name}" <${email.address}>` : email.address
       }));
     });
-  }
+  };
 
-  render() {
-    const {handleSubmit, invitees, inviteesRaw, submitting, teamName} = this.props;
+  const onInviteTeamSubmit = async data => {
+    const serverInvitees = data.invitees.map(invitee => {
+      // Remove label field:
+      const {label, ...inviteeForServer} = invitee;
+      return inviteeForServer;
+    });
+    const options = {
+      variables: {
+        teamId,
+        invitees: serverInvitees
+      }
+    };
 
-    const invitesFieldHasError = false; // TODO: wire this up for real
-    const helpText = invitesFieldHasError ?
-      // eslint-disable-next-line max-len
-      <span>Oops! Please make sure email addresses are valid <br />and separated by a single comma.</span> :
-      <span>You can paste multiple emails separated by a comma.<br />&nbsp;</span>;
+    const res = await cashay.mutate('inviteTeamMembers', options);
```

Comment:
I mean, this reads to nicely. I don't think you should feel bad about the Promise interface here. It looks great!